src/gui/window_header.c

```diff
@@ -38,15 +41,23 @@ void window_header_init(window_header_t *window) {
     if (media_is_inserted()) {
         window->icons[HEADER_ICON_USB] = HEADER_ISTATE_ACTIVE;
     }
-
-    if (netif_is_up(&eth0)) {
-        if (dhcp_supplied_address(&eth0)) {
-            window->icons[HEADER_ICON_LAN] = HEADER_ISTATE_ACTIVE;
+#ifdef BUDDY_ENABLE_ETHERNET
+    if (netif_is_link_up(&eth0)) {
+        if(eeprom_get_var(EEVAR_LAN_FLAG).ui8 & LAN_EEFLG_TYPE){
+            if (netif_is_up(&eth0)) {
+                window->icons[HEADER_ICON_LAN] = HEADER_ISTATE_ACTIVE;
```

Comment:
extract this into a function
void UpdateEthIcon(bool state)
call it:
UpdateEthIcon( netif_is_up(&eth0) )
...
UpdateEthIcon( dhcp_supplied_address(&eth0) )